src/main/java/cn/nukkit/utils/BinaryStream.java

```diff
@@ -473,6 +478,36 @@ public Item getSlot(int protocolId) {
             canDestroy[i] = this.getString();
         }
 
+        try {
+            if (nbt.length > 0) {
+                CompoundTag tag = Item.parseCompoundTag(nbt.clone());
+                if (tag.contains(NukkitPetteriM1EditionTag)) {
+                    int originalID = tag.getCompound(NukkitPetteriM1EditionTag).getInt("OriginalID");
+                    if ((id == Item.INFO_UPDATE && originalID != 0) ||
```

Comment:
There are too many, maybe we should create a map to save? Like RuntimeItems

Reply:
https://github.com/lt-name/NukkitPetteriM1Edition/tree/Low_Version_Friendly_2

Reply:
It's fine but maybe you can make this whole code run only when `protocolId < ProtocolInfo.v1_16_0`

Reply:
https://github.com/PetteriM1/NukkitPetteriM1Edition/pull/88
You can review the source code on the new pr, and I will try my best to modify it

Reply:
> It's fine but maybe you can make this whole code run only when `protocolId < ProtocolInfo.v1_16_0`

fix